### PR TITLE
Remove lender from loan

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: contracts/truefi2/deprecated/LegacyLoanToken2.sol
Comment (yuchenlintt) [inline contracts/truefi2/deprecated/LegacyLoanToken2.sol:160]: We should guard this just in case we deploy this again.

```suggestion
    function setLender(address _lender) external onlyAdmin {
```

**Review by yuchenlintt — DISMISSED** [review]
LGTM, but let's wait until #1019 is merged to see if that helps fix `test-truefi2` for this PR.

File: contracts/truefi2/deprecated/LegacyLoanToken2.sol
Comment (vanruch) [reply contracts/truefi2/deprecated/LegacyLoanToken2.sol:160]: This contract has some changes to ease testing, like removing onlyLiquidator modifier. We should not deploy it ever.

Comment (yuchenlintt) [reply contracts/truefi2/deprecated/LegacyLoanToken2.sol:160]: I see, I misinterpreted the intent of `LegacyLoanToken2` then. If this contract is intended only for use in tests, then let's move this contract to the `mocks/` folder and rename it to `TestLoanToken2`, `TestLegacyLoanToken2`, or something similar. (Can be discussed and done in a separate PR.)

I thought the intent of the `deprecated/` subdir was for contracts that we need to keep for compatibility and in case of urgent upgrade. But if that wasn't the intent then my bad for misinterpreting.

**Review by yuchenlintt — DISMISSED** [review]

File: contracts/truefi2/LoanFactory2.sol
Comment (skalermo) [inline contracts/truefi2/LoanFactory2.sol:40]:
```suggestion
    address private DEPRECATED__lender;
```

Comment (vanruch) [reply contracts/truefi2/LoanFactory2.sol:40]: isn't it the same thing?

Comment (vanruch) [reply contracts/truefi2/LoanFactory2.sol:40]: Ok, it's not. They are internal by default

**Review by skalermo — APPROVED** [review]

File: test/truefi2/Liquidator2.test.ts
Comment (MiksuJak) [inline test/truefi2/Liquidator2.test.ts:499]: good catch

**Review by MiksuJak — APPROVED** [review]

**Pull request merged**
